File: .idea/app/src/main/java/com/chevstrap/rbx/UI/ViewModels/Settings/BehaviourViewModel.java
package com.chevstrap.rbx.UI.ViewModels.Settings;

import com.chevstrap.rbx.App;
import com.chevstrap.rbx.ConfigManager;
import com.chevstrap.rbx.Enums.RobloxAppType;
import com.chevstrap.rbx.R;
import com.chevstrap.rbx.UI.Frontend;

import java.util.Map;
import java.util.Objects;

public class BehaviourViewModel {
    public BehaviourViewModel() {}

    public boolean isEnableBringToLatestUpdate() {
        return App.getChevstrapSettings().isManualUpdateBootstrapper();
    }

    public void setEnableBringToLatestUpdate(boolean value) {
        App.getChevstrapSettings().setManualUpdateBootstrapper(value);
    }

    public String get_preferred_roblox_app() {
        String value = App.getChevstrapSettings().getPreferredRobloxApp();
        for (Map.Entry<RobloxAppType, String> entry : ConfigManager.RobloxAppTypes.entrySet()) {
            if (Objects.equals(entry.getValue(), value)) {
                return entry.getKey().name();
            }
        }
        return ConfigManager.RobloxAppTypes.keySet().iterator().next().name();
    }

    public void set_preferred_roblox_app(String modeName) {
        if (modeName == null) {
            return;
        }

        try {
            RobloxAppType mode = RobloxAppType.valueOf(modeName);
            App.getChevstrapSettings().setPreferredRobloxApp(ConfigManager.RobloxAppTypes.get(mode));
        } catch (IllegalArgumentException e) {
            App.getChevstrapSettings().setPreferredRobloxApp("global");
        }
    }
}

File: app/src/main/java/com/chevstrap/rbx/ui/viewModels/settings/pages/BehaviourViewModel.kt
package com.chevstrap.rbx.ui.viewModels.settings.pages

import com.chevstrap.rbx.App
import com.chevstrap.rbx.ConfigManager
import com.chevstrap.rbx.enums.LaunchMode

class BehaviourViewModel {

    var bringToLatestUpdate: Boolean
        get() = App.config.data.manualUpdateBootstrapper
        set(value) {
            App.config.data.manualUpdateBootstrapper = value
        }

    var preferredRobloxApp: String
        get() {
            val value = App.config.data.preferredRobloxApp

            for (entry in ConfigManager.RobloxAppTypes.entries) {
                if (entry.value == value) {
                    return entry.key.name
                }
            }

            return ConfigManager.RobloxAppTypes.keys
                .firstOrNull()
                ?.name
                ?: LaunchMode.Global.name
        }

        set(modeName) {
            try {
                val mode = LaunchMode.valueOf(modeName)
                App.config.data.preferredRobloxApp =
                    ConfigManager.RobloxAppTypes[mode] ?: "global"
            } catch (_: IllegalArgumentException) {
                App.config.data.preferredRobloxApp = "global"
            }
        }
}